src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -89,9 +104,18 @@ class AppsResource @Inject() (
     async {
       implicit val identity = await(authenticatedAsync(req))
 
-      val rawApp = Raml.fromRaml(Json.parse(body).as[raml.App].normalize)
+      val rawApp = Raml.fromRaml(Json.parse(body).as[raml.App].normalize(validateAndNormalizeApp))
       val now = clock.now()
-      val app = validateOrThrow(rawApp).copy(versionInfo = VersionInfo.OnlyVersion(now))
+
+      // This is not really thread safe, another thread may intercept us and change the enforceRole flag, so we need
+      // to revalidate this inside the the groupManager later
+      val roleEnforcement = getEnforcedRoleForService(rawApp.id)
```

Comment:
What do you think about using a `Validator[Role]` or something similar instead? It could still be constructed dending on `group.enforceRole` but would still be in line with the validation we use.